.github/workflows/docker-images.yml

```diff
@@ -53,31 +59,45 @@ jobs:
       - name: Build Image
         if: steps.should-run.outputs.result == 'true'
         run: |-
-          docker build                                                          \
-            --pull                                                              \
-            --build-arg BUILD_TIMESTAMP="$(date -u +'%Y-%m-%dT%H:%M:%SZ')"      \
-            --build-arg COMMIT_ID='${{ github.sha }}'                           \
-            --tag 'jsii/superchain:nightly'                                     \
+          docker build                                                        \
+            --pull                                                            \
+            --build-arg BUILD_TIMESTAMP="$(date -u +'%Y-%m-%dT%H:%M:%SZ')"    \
+            --build-arg COMMIT_ID='${{ github.sha }}'                         \
+            --build-arg NODE_MAJOR_VERSION=${{ matrix.node }}                 \
+            --tag "jsii/superchain:node${{ matrix.node }}-nightly"            \
             ./superchain
       - name: Test Image
         if: steps.should-run.outputs.result == 'true'
         run: |-
-          docker run                                                            \
-            --rm                                                                \
-            --tty                                                               \
-            --network=host                                                      \
-            -v${{ github.workspace }}:${{ github.workspace }}                   \
-            -w${{ github.workspace }}                                           \
-            'jsii/superchain:nightly'                                           \
+          docker run                                                          \
+            --rm                                                              \
+            --tty                                                             \
+            --network=host                                                    \
+            -v${{ github.workspace }}:${{ github.workspace }}                 \
+            -w${{ github.workspace }}                                         \
+            "jsii/superchain:node${{ matrix.node }}-nightly"                  \
```

Comment:
You could have left `'` here - the `${{}}` stuff gets interpolated *before* the job runs.